app/components/agenda/agendaitem/agendaitem-case-panel/agendaitem-case-panel-edit.js

```diff
@@ -154,4 +157,25 @@ export default class AgendaitemCasePanelEdit extends Component {
   pasteIntoTitle = (pasteEvent) => {
     this.args.agendaitem.title = cleanPasteInputForTextarea(pasteEvent, 'title-agendaitem', this.args.agendaitem.title);
   }
+
+  rollbackDirtyAttributes = () => {
+    if (this.args.agendaitem.hasDirtyAttributes) {
+      this.args.agendaitem.rollbackAttributes();
+    }
+    // We change the value of confidental directly on subcase, so we should also roll it back
+    if (this.args.subcase?.hasDirtyAttributes) {
+      this.args.subcase.rollbackAttributes();
+    }
+    if (this.newsItem?.hasDirtyAttributes) {
+      this.newsItem.rollbackAttributes();
+    }
+    if (this.internalReview?.hasDirtyAttributes) {
+      this.internalReview.rollbackAttributes();
+    }
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.rollbackDirtyAttributes();
```

Comment:
I think we need a `this.preventUnload.disable();` here.

Right now the following doesn't work as intended:
1. Open an agendaitem and edit the subcase properties
2. Click on a different agendaitem -> no warning
3. Try to navigate to a different route -> warning